scipy/stats/morestats.py

```diff
@@ -32,11 +33,22 @@ def __dir__():
 def __getattr__(name):
     if name not in __all__:
         raise AttributeError(
-            "scipy.stats.morestats is deprecated and has no attribute "
-            f"{name}. Try looking in scipy.stats instead.")
-
-    warnings.warn(f"Please use `{name}` from the `scipy.stats` namespace, "
-                  "the `scipy.stats.morestats` namespace is deprecated.",
-                  category=DeprecationWarning, stacklevel=2)
+            f"`scipy.stats.morestats` has no attribute `{name}`; furthermore, "
+            "`scipy.stats.morestats` is deprecated and will be removed in "
+            "SciPy 2.0.0.")
+
+    attr = getattr(scipy.stats, name, None)
+
+    if attr is not None:
+        message = (f"Please import `{name}` from the `scipy.stats` namespace; "
+                   "the `scipy.stats.morestats` namespace is deprecated and "
+                   "will be removed in SciPy 2.0.0.")
+    else:
+        message = (f"`scipy.stats.morestats.{name}` is deprecated along with "
+                   "the `scipy.stats.morestats` namespace. "
+                   f"`scipy.stats.morestats.{name}` will be removed in SciPy 1.13.0, and "
```

Comment:
```suggestion
                   f"`scipy.stats.morestats.{name}` will be removed in SciPy 1.14.0, and "
```

Reply:
So, I think this reaction is to our usual "2 versions from deprecation warning to removal" policy, but even though it's not obvious from the diff in this case, all(?) of these are raising deprecation warnings already, and we're only sharpening it. This is in line with how we've done this for other long-existing deprecations, and I'm fine to announce removal for 1.13.

```
>>> import scipy
>>> scipy.__version__
'1.11.1'
>>> from scipy.stats.morestats import wilcoxon
<stdin>:1: DeprecationWarning: Please use `wilcoxon` from the `scipy.stats` namespace, the `scipy.stats.morestats` namespace is deprecated.
>>> from scipy.stats import wilcoxon
>>> from scipy.stats.morestats import WilcoxonResult
<stdin>:1: DeprecationWarning: Please use `WilcoxonResult` from the `scipy.stats` namespace, the `scipy.stats.morestats` namespace is deprecated.
>>> from scipy.stats import WilcoxonResult
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: cannot import name 'WilcoxonResult' from 'scipy.stats' (C:\Users\axelo\.conda\envs\dev\lib\site-packages\scipy\stats\__init__.py)
```

Reply:
Good point